### DISCUSSION TITLE
Selecting and deleting multiple rows in a datagrid

### DISCUSSION THREAD

**Discussion by RiseAdministrative77** [AvaloniaUI/Avalonia#6265] [Q&A] [answered]
I'd like to be able to use the already present multi-row selection in datagrid to delete multiple rows at a time. 
I'd rather use this than solutions like one checkbox/button per row.
I've tried sending _SelectedItems_ as a _commandparameter_ but couldn't cast as _datagridselecteditemscollection_ (as it is internal), __selectedItemsCache_ field is private and I can't even seem to extract items with reflection. (I can see my selected items while debugging though but there might be some debugger magic there)
Another path would've been to use _CanUserDeleteRows_ but it doesn't seem to exist on Avalonia.
What are my options, if any?
Did I miss something?

**Comment by EtherGhost** [comment, marked as answer]
Have you tried to bind SelectedItems to a ObserverableCollection Property on the ViewModel?
Then you should be able to use them to remove from you ItemsSource ObservebleCollection

**Reply by maxkatz6** [reply]
Yes, operating over ObserverableCollection, which is bind to the datagrid, in overall is the right way.

**Reply by RiseAdministrative77** [reply]
Datagrid's SelectedItems is neither bindable from xaml nor from codebehind (Avalonia 0.10.6). 
It's not proposed as a property by the editor and it throws an exception (System.ArgumentException : 'The item is not contained in the ItemsSource. Arg_ParamName_Name') if I try to bind anyway.

**Reply by maxkatz6** [reply]
I don't understand your problem and what are you trying to achive with these approaches.
If you have observable collection bind to the Items, you can delete multiple items from there.
You can read SelectedItems, but you will need to cast it to your array type, using ```OfType<MyItem>()``` as an example (why would you ever need internal datagridselecteditemscollection for that?).

You also can subscribe on SelectionChanged and update second observable collection of items, selected items. It also would be nice to be bindable. But neither is needed in your case.

**Reply by RiseAdministrative77** [reply]
You made me realize SelectedItems was actually an IList and I just have to cast the right way (as you said, using OfType<T>()).
I was very confused because either from CommandParameter or code behind, debugger would indicate SelectedItems as a datagridselecteditemscollection.
Anyway, problem solved, used CommandParameter of type IList and I can delete everything now! Thanks a lot!

**Reply by maxkatz6** [reply]
No worries. DataGrid isn't the best example of API either, because it was ported from Silverlight and can be a bit confusing in places (like, SelectedItems is not bindable).

**Reply by EtherGhost** [reply]
SelectedItems really should be bindable. It's this something that will be implemented?

**Reply by maxkatz6** [reply]
Maybe. If somebody from community will do it.
Not planned by the core team.
